managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2108,17 +2111,7 @@ void doCacheEviction(long maxTimestamp) {
     }
 
     private PositionImpl getEarlierReadPositionForActiveCursors() {
-        PositionImpl smallest = null;
-        for (ManagedCursor cursor : activeCursors) {
-            PositionImpl p = (PositionImpl) cursor.getReadPosition();
-            if (smallest == null) {
-                smallest = p;
-            } else if (p.compareTo(smallest) < 0) {
-                smallest = p;
-            }
-        }
-
-        return smallest;
+        return (PositionImpl) (activeCursors.getSlowestReader()).getReadPosition();
```

Comment:
there's also a method called `ManagedCursorContainer.getSlowestReaderPosition()`, could that have been used?

Reply:
it looks that is uses `getMarkDeletedPosition()` so it's probably different than `activeCursors.getSlowestReader().getReadPosition()`?

Reply:
You are right, now it seems to ignore the non-durable cursor